### PR TITLE
Upgrade to Electron 22

### PR DESCRIPTION
**Pull request by gre**
### 📝 Description

- **Electron was upgraded from 15 to 22. (major)** as well as "electron-builder"
  - that means we have a new bump of the nodejs version for native module too. So all native library impacted (coins, usb,..)
  - that means we have a new Chromium version. So the UI is impacted. Screenshots have changed due to this.
  - WebView usage of Discover section was adapted to make external link opening to work.
- Sentry was upgraded (minor)

replaces https://github.com/LedgerHQ/ledger-live/pull/1463

latest builds ( on d16b88786af4f1b57ac358e6615852249f7f2845 )

https://download.live.ledger.com/ledger-live-desktop-2.52.0-electron22.0-mac.dmg
https://download.live.ledger.com/ledger-live-desktop-2.52.0-electron22.0-win-x64.exe
https://download.live.ledger.com/ledger-live-desktop-2.52.0-electron22.0-linux-x86_64.AppImage

### ❓ Context

- **Impacted projects**: `LLD` 
- **Linked resource(s)**: LIVE-4331 
  - fixes https://github.com/LedgerHQ/ledger-live/issues/1005
  
### ✅ Checklist

- [x] **Test coverage** covered by playwright tests but non regression on native will be needed for all OSs 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

✅  Sentry is still working as expected 

Screenshot 2022-12-15 at 16 56 47

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

**Comment by gre** [comment]
✅  expect some "flawkyness" cases there are no regression I could see in screenshots, so all look good in term of UI.

I think we will soon be ready for QA.

**Comment by Justkant** [comment]
When or after rebasing, we'll need to duplicate the solution for the new WebViewV2 using the Wallet API
I can help for that

**Comment by gre** [comment]
@Justkant ah zut :( du coup ça va encore décaler Electron 22. tu peux envoyer une PR?

**Comment by Justkant** [comment]
@gre On it :+1:

**Comment by Justkant** [comment]
PR available here: https://github.com/LedgerHQ/ledger-live/pull/2182

**Comment by gre** [comment]
builds
- https://download.live.ledger.com/ledger-live-desktop-2.52.0-electron22.0-mac.dmg
- https://download.live.ledger.com/ledger-live-desktop-2.52.0-electron22.0-win-x64.exe
- https://download.live.ledger.com/ledger-live-desktop-2.52.0-electron22.0-linux-x86_64.AppImage

File: apps/ledger-live-desktop/src/main/index.js [hunk 1]
```diff
@@ -158,6 +158,19 @@ app.on("ready", async () => {
 
   ipcMain.on("log", (event, { log }) => logger.log(log));
 
+  // To handle openning new windows from webview
```
Comment (juan-cortes) [inline apps/ledger-live-desktop/src/main/index.js:161]:
```suggestion
  // To handle opening new windows from webview
```

File: apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/WebView.tsx [hunk 2]
```diff
@@ -389,11 +402,25 @@ export function WebView({ manifest, onClose, inputs = {}, config }: Props) {
...
+        <webview
           src={url.toString()}
           ref={targetRef}
-          style={{ opacity: widgetLoaded ? 1 : 0 }}
+          /**
+           * There seem to be an issue between Electron webview and styled-components
+           * (and React more broadly, cf. comment bellow).
+           * When using a styled webview componennt, the `allowpopups` prop does not
+           * seem to be set
+           */
```
Comment (juan-cortes) [inline apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/WebView.tsx:413]:
```suggestion
          /**
           * There seems to be an issue between Electron webview and styled-components
           * (and React more broadly, cf. comment below).
           * When using a styled webview component, the `allowpopups` prop does not
           * seem to be set
           */
```

File: apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/WebView.tsx [hunk 3]
```diff
@@ -389,11 +402,25 @@ export function WebView({ manifest, onClose, inputs = {}, config }: Props) {
...
+          style={webviewStyle}
           preload={`file://${remote.app.dirname}/webviewPreloader.bundle.js`}
+          /**
+           * There seems to be an issue between Electron webview and react
+           * Hence, the normal `allowpopups` prop does not work and we need to
+           * explicitly set it's value to "true" as a string
+           * cf. https://github.com/electron/electron/issues/6046
```
Comment (juan-cortes) [inline apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/WebView.tsx:420]:
```suggestion
           * There seems to be an issue between Electron webview and react
           * Hence, the normal `allowpopups` prop does not work and we need to
           * explicitly set its value to "true" as a string
           * cf. https://github.com/electron/electron/issues/6046
```

File: apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/WebViewV2.tsx [hunk 4]
```diff
@@ -485,31 +485,44 @@ export function WebView({ manifest, onClose, inputs = {}, config }: Props) {
...
+
+  const webviewStyle = useMemo(() => {
+    return {
+      opacity: widgetLoaded ? 1 : 0,
+      border: "none",
+      width: "100%",
+      flex: 1,
+      transition: "opacity 200ms ease-out",
+    };
+  }, [widgetLoaded]);
```
Comment (juan-cortes) [inline apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/WebViewV2.tsx:525]: I lack context as to why we have two versions of this.
Can't we kill one of them already?

File: apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/WebViewV2.tsx [hunk 5]
```diff
@@ -523,11 +536,25 @@ export function WebView({ manifest, onClose, inputs = {}, config }: Props) {
...
+        <webview
           src={url.toString()}
           ref={targetRef}
-          style={{ opacity: widgetLoaded ? 1 : 0 }}
+          /**
+           * There seem to be an issue between Electron webview and styled-components
+           * (and React more broadly, cf. comment bellow).
+           * When using a styled webview componennt, the `allowpopups` prop does not
+           * seem to be set
+           */
```
Comment (juan-cortes) [inline apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/WebViewV2.tsx:547]:
```suggestion
          /**
           * There seem to be an issue between Electron webview and styled-components
           * (and React more broadly, cf. comment below).
           * When using a styled webview componennt, the `allowpopups` prop does not
           * seem to be set
           */
```

File: apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/WebViewV2.tsx [hunk 6]
```diff
@@ -523,11 +536,25 @@ export function WebView({ manifest, onClose, inputs = {}, config }: Props) {
...
+          style={webviewStyle}
           preload={`file://${remote.app.dirname}/webviewPreloader.bundle.js`}
+          /**
+           * There seems to be an issue between Electron webview and react
+           * Hence, the normal `allowpopups` prop does not work and we need to
+           * explicitly set it's value to "true" as a string
+           * cf. https://github.com/electron/electron/issues/6046
+           */
```
Comment (juan-cortes) [inline apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/WebViewV2.tsx:555]:
```suggestion
          /**
           * There seems to be an issue between Electron webview and react
           * Hence, the normal `allowpopups` prop does not work and we need to
           * explicitly set its value to "true" as a string
           * cf. https://github.com/electron/electron/issues/6046
           */
```

**Review by juan-cortes — CHANGES_REQUESTED** [review]
Mostly just grammar stuff, but there seems to be something going on with the outline of the selectors on the screenshots. Inconsistently making them blue/purple. I don't mind either color but if they're inconsistent they will break CI on all prs again, and we don't want to go back to flaky tests.

image

**Comment by gre** [comment]
@juan-cortes the behavior of the border is not inconsistent in this PR. it's just a new state and it's apparently a playwright bug.

File: apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/WebViewV2.tsx (hunk 4 above)
Comment (gre) [reply apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/WebViewV2.tsx:525]: it's a good point but this is unrelated to Electron 22, we just had to make sure both works still properly. ( cc @Justkant  )
this feedback need to be taken somewhere else (unrelated to this PR to "kill one of them")

Comment (Justkant) [reply apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/WebViewV2.tsx:525]: We just introduced `WebViewV2` in order to support the new wallet-api while not touching the old platform stuff for the `live-app-sdk`

**Review by Justkant — APPROVED** [review]

**Review by juan-cortes — APPROVED** [review]
letz go!

**Pull request merged**